src/printer_tree.c

```diff
@@ -0,0 +1,3107 @@
+    } else if (ind_node1.indent.type == TRD_INDENT_IN_NODE_FAILED) {
+        /* node name is too long */
+        trp_print_line_up_to_node_name(node, ipck.wrapper, out);
+        if (trp_node_body_is_empty(node)) {
+            return;
+        } else {
+            ly_print_(out, "\n");
+            ind_node2 = trp_second_half_node(node, ind_node1.indent);
+            ind_node2.indent.type = TRD_INDENT_IN_NODE_DIVIDED;
+            tmp = (struct trt_pck_indent) {
+                trp_wrapper_if_last_sibling(ipck.wrapper, node.last_one), ind_node2.indent
+            };
+            trp_print_divided_node(ind_node2.node, ppck, tmp, mll, out);
+        }
+
+    }
+}
+
+/******************************************************************************
+ * Definition of Tro reading functions
+ *****************************************************************************/
+
+/**
+ * @brief Return trt_parent_cache filled with default values.
+ */
+static struct trt_parent_cache
+tro_empty_parent_cache()
+{
+    return (struct trt_parent_cache) {
+               TRD_ANCESTOR_ELSE,
+               LYS_STATUS_CURR,
+               LYS_CONFIG_W,
+               NULL
+    };
+}
+
+/**
+ * @brief Get new trt_parent_cache if we apply the transfer to the child node in the tree.
+ * @param[in] ca is parent cache for current node.
+ * @param[in] pn is pointer to the current tree node.
+ * @return Cache for the current node.
+ */
+static struct trt_parent_cache
+tro_parent_cache_for_child(struct trt_parent_cache ca, const struct lysp_node *pn)
+{
+    struct trt_parent_cache ret = {};
+
+    ret.ancestor =
+            pn->nodetype & (LYS_INPUT) ? TRD_ANCESTOR_RPC_INPUT :
+            pn->nodetype & (LYS_OUTPUT) ? TRD_ANCESTOR_RPC_OUTPUT :
+            pn->nodetype & (LYS_NOTIF) ? TRD_ANCESTOR_NOTIF :
+            ca.ancestor;
+
+    ret.lys_status =
+            pn->flags & (LYS_STATUS_CURR | LYS_STATUS_DEPRC | LYS_STATUS_OBSLT) ? pn->flags :
+            ca.lys_status;
+
+    ret.lys_config =
+            ca.ancestor == TRD_ANCESTOR_RPC_INPUT ? 0 :     /* because <flags> will be -w */
+            ca.ancestor == TRD_ANCESTOR_RPC_OUTPUT ? LYS_CONFIG_R :
+            pn->flags & (LYS_CONFIG_R | LYS_CONFIG_W) ? pn->flags :
+            ca.lys_config;
+
+    ret.last_list =
+            pn->nodetype & (LYS_LIST) ? (struct lysp_node_list *)pn :
+            ca.last_list;
+
+    return ret;
+}
+
+/**
+ * @brief Read next sibling of the current node.
+ * @param[in] origin_tc is context of the tree.
+ * @return A patch structure that has prepared pointers for updating struct trt_tree_ctx.
+ *  If sibling exists then .pn points to him, otherwise is set to NULL.
+ *  The .tpn points to its sibling if it exists and if .pn points to the same node as .tpn,
+ *  otherwise .tpn value from origin_tc is copied.
+ */
+static struct trt_tree_ctx_node_patch
+tro_read_next_sibling(const struct trt_tree_ctx *origin_tc)
+{
+    assert(origin_tc != NULL && origin_tc->pn != NULL);
```

Comment:
Generally, you can leave out `!= NULL`, it is often used that way for checking parameters. But you can use it for any NULL pointer comparison.